Realm/RLMRealm.h

```diff
@@ -495,6 +496,22 @@ __deprecated_msg("Use `performMigrationForConfiguration:error:`") NS_REFINED_FOR
  */
 + (BOOL)performMigrationForConfiguration:(RLMRealmConfiguration *)configuration error:(NSError **)error;
 
+/**
+ Creates a `RLMThreadHandover` from the passed in objects that can be safely imported on a separate thread.
+
+ Used to pass objects between threads in a thread-safe manner.
+
+ @warning Note that the returned `ThreadHandover` must be imported at most once, and that the current version of the
+ Realm will remain pinned until this package is imported or deallocated.
+
+ @param objects The objects to pass between threads.
+ @return A package that can be imported on a separate thread.
+
+ @see RLMThreadHandover
+ */
+- (RLMThreadHandover *)exportThreadHandoverWithObjects:(NSArray<id<RLMThreadConfined>> *)objects
```

Comment:
I like the API rethink, renaming `Realm.threadSafeLookup(_:)` to `ThreadSafeReference.resolve(in:)`.

```swift
let fooRef = foo.threadSafeReference
let barRef = bar.threadSafeReference
queue.async {
    let realm = try! Realm()
    let foo = fooRef.resolve(in: realm) // correct type!
    let bar = barRef.resolve(in: realm) // correct type!
    // do stuff
}
```

```swift
let references = [foo, bar, baz].map { $0.threadSafeReference }
queue.async {
    let realm = try! Realm()
    let items = references.map { $0.resolve(in: realm) }
    // do stuff
}
```